src/prefect/agent/kubernetes/agent.py

```diff
@@ -262,36 +262,41 @@ def manage_jobs(self) -> None:
                             # Format pod failure error message
                             failed_pods.append(pod.metadata.name)
                             pod_status_logs = [f"Pod {pod.metadata.name} failed."]
-                            for status in pod.status.container_statuses:
-                                state = (
-                                    "running"
-                                    if status.state.running
-                                    else "waiting"
-                                    if status.state.waiting
-                                    else "terminated"
-                                    if status.state.terminated
-                                    else "Not Found"
-                                )
+                            if not pod.status.container_statuses:
                                 pod_status_logs.append(
-                                    f"\tContainer '{status.name}' state: {state}"
+                                    "\tNo container statuses found for pod"
                                 )
-
-                                if status.state.terminated:
+                            else:
+                                for status in pod.status.container_statuses:
+                                    state = (
+                                        "running"
+                                        if status.state.running
+                                        else "waiting"
+                                        if status.state.waiting
+                                        else "terminated"
+                                        if status.state.terminated
+                                        else "Not Found"
```

Comment:
This took me a second to parse as valid python, I hadn't ever done multi condition if-else chain. Good to know.